src/app/components/admin/all-uploads/all-uploads.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnInit } from "@angular/core";
+import { PageComponent } from "@helpers/page/pageComponent";
+import { List } from "immutable";
+import { ShallowHarvestsService } from "@baw-api/harvest/harvest.service";
+import { Filters } from "@baw-api/baw-api.service";
+import { Harvest } from "@models/Harvest";
+import {
+  BehaviorSubject,
+  Observable,
+  catchError,
+  takeUntil,
+  throwError,
+} from "rxjs";
+import { CLIENT_TIMEOUT } from "@baw-api/api.interceptor.service";
+import { BawApiError } from "@helpers/custom-errors/baw-api-error";
+import { NgbModal } from "@ng-bootstrap/ng-bootstrap";
+import { adminCategory, adminUploadsMenuItem } from "../admin.menus";
+import { adminMenuItemActions } from "../dashboard/dashboard.component";
+
+@Component({
+  selector: "baw-all-uploads",
+  templateUrl: "all-uploads.component.html",
+})
+class AllUploadsComponent extends PageComponent implements OnInit {
+  public constructor(
+    public modals: NgbModal,
+    private api: ShallowHarvestsService
+  ) {
+    super();
+  }
+
+  public filters$: BehaviorSubject<Filters<Harvest>>;
+
+  public ngOnInit(): void {
+    this.filters$ = new BehaviorSubject({
+      sorting: {
+        direction: "desc",
+        orderBy: "createdAt",
+      },
+      // projection allows us only to emit the fields that we want
+      // this improves performance and reduces the amount of data sent
+      projection: {
+        include: [
+          "id",
+          "projectId",
+          "name",
+          "createdAt",
+          "creatorId",
+          "streaming",
+          "status",
+        ],
+      },
+    });
+  }
+
+  public getModels = (filters: Filters<Harvest>): Observable<Harvest[]> =>
+    this.api.filter(filters);
+
+  public async abortUpload(template: any, harvest: Harvest): Promise<void> {
+    const ref = this.modals.open(template);
+    const success = await ref.result.catch((_) => false);
+
+    if (success) {
+      this.api
+        .transitionStatus(harvest, "complete")
+        .pipe(
+          catchError((err: BawApiError) => {
+            if (err.status !== CLIENT_TIMEOUT) {
+              return throwError(() => err);
+            }
+          })
+        )
+        .pipe(takeUntil(this.unsubscribe))
+        .subscribe({
+          next: (): void => {
+            this.filters$.next({});
+          },
+          error: (err: BawApiError): void => {
+            this.notifications.error(err.message);
+          },
+        });
+    }
+  }
+}
+
```

Comment:
i reckon this function is identical to the one in the non-admin view? that's probably a bit too much duplication...